File: cgi-bin/read_file.php
<?php
$fileContents = file_get_contents($_FILES['file']['tmp_name']);

$body = "<!DOCTYPE html>
<html>
<head>
<title>Received POST Data</title>
</head>
<body>
<h1>Info passed by stdin was:</h1>
<p>$fileContents</p>
</body>
</html>";

header("Content-type: text/html; charset=UTF-8");

echo $body;
?>

File: cgi-bin/read_file.py
#!/usr/bin/python3
import os
import sys
import cgi
import cgitb

cgitb.enable()

try:
    form = cgi.FieldStorage(keep_blank_values=True)
    fileContents = form['file'].file.read().decode('utf-8')

    print ("Content-Type: text/html; charset=UTF-8")

    print()

    print ('''<!DOCTYPE html>
    <html>
    <head>
    <title>Read FILE</title>
    </head>
    <body>
    <h1>File content is:</h1>
    <pre> ''')
    print (fileContents)
    print ('''<pre>
    </body>
    </html>''')

    sys.exit()

except Exception as e:
    print("Status: 500 Internal Server Error")
